addons/account/models/account_bank_statement.py

```diff
@@ -115,7 +115,7 @@ def validate(self):
 class AccountBankStatement(models.Model):
     _name = "account.bank.statement"
     _description = "Bank Statement"
-    _order = "date desc, name desc, id desc"
+    _order = "date desc, id desc, name desc"
```

Comment:
Why changing the order? This will completely change the display. From a usability point of view, date and name is better than date, id.

If the problem is for people not using the journal sequence and adding their own name for the bank statement while in draft. Maybe change the method used to compute the previous_statement_id to take the name into account? Dunno if this will solve the problem but it's worth a try.

```diff
-@api.depends('date', 'journal_id')
+@api.depends('date', 'journal_id', 'name')
    def _get_previous_statement(self):
        for st in self:
            # Search for the previous statement
            domain = [('date', '<=', st.date), ('journal_id', '=', st.journal_id.id)]
+           if self.name:
+               domain.extend([('name', '<', self.name)])
            # The reason why we have to perform this test is because we have two use case here:
            # First one is in case we are creating a new record, in that case that new record does
            # not have any id yet. However if we are updating an existing record, the domain date <= st.date
            # will find the record itself, so we have to add a condition in the search to ignore self.id
            if not isinstance(st.id, models.NewId):
                domain.extend(['|', '&', ('id', '<', st.id), ('date', '=', st.date), '&', ('id', '!=', st.id), ('date', '!=', st.date)])
            previous_statement = self.search(domain, limit=1)
            st.previous_statement_id = previous_statement.id
```

Reply:
@csnauwaert Why not doing this?